### PR TITLE
bugfix(libs/header): store incoming head in syncer

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
Closes https://github.com/celestiaorg/celestia-node/issues/1728

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: libs/header/sync/sync.go [hunk 1]
```diff
@@ -41,8 +42,8 @@ type Syncer[H header.Header] struct {
 	state   State
 	// signals to start syncing
 	triggerSync chan struct{}
-	// syncedHead is the latest synced header.
-	syncedHead H
+	// syncedHeadPtr is the latest synced header.
+	syncedHeadPtr atomic.Pointer[H]
```
Comment (distractedm1nd) [inline libs/header/sync/sync.go:46]: optional nit, but I think we should keep the name - the type information that it's a pointer is already in the type

**Review by distractedm1nd — APPROVED** [review]
approving because my nit is optional, logic is good

File: libs/header/sync/sync_head.go [hunk 2]
```diff
@@ -112,6 +112,7 @@ func (s *Syncer[H]) incomingNetHead(ctx context.Context, netHead H) pubsub.Valid
 	_, err := s.store.Append(ctx, netHead)
 	if err == nil {
 		// a happy case where we appended maybe head directly, so accept
+		s.syncedHeadPtr.Store(&netHead)
```
Comment (renaynay) [inline libs/header/sync/sync_head.go:115]: nice find.

File: libs/header/sync/sync.go [hunk 3]
```diff
@@ -177,42 +178,47 @@ func (s *Syncer[H]) sync(ctx context.Context) {
 		return
 	}
 
-	if s.syncedHead.IsZero() {
+	headPtr := s.syncedHeadPtr.Load()
```
Comment (renaynay) [inline libs/header/sync/sync.go:181]:
```suggestion
	headPtr := s.syncedHeadPtr.Load()
	
```

File: libs/header/sync/sync.go (hunk 1 above)
Comment (vgonkivs) [reply libs/header/sync/sync.go:46]: Ok. Will change it

**Review by renaynay — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**